src/main/java/com/example/demo/repository/dao/CaseFileDao.java

```diff
@@ -21,9 +21,25 @@ public CaseFileDao(EntityManager entityManager) {
         this.entityManager = entityManager;
     }
 
-    public Optional<CaseFile> create(CaseFile caseFile) {
+    public Optional<CaseFileDTO> create(CaseFileDTO caseFileDTO) {
+        if (caseFileDTO == null) {
+            return Optional.empty();
+        }
+        Person person;
+        Long personId = caseFileDTO.getPersonId();
+
+        if (personId != null) {
+            person = entityManager.find(Person.class, personId);
+        } else {
+            person = EntityDTOConverter.convertToPersonEntity(caseFileDTO.getPersonDTO());
+            entityManager.persist(person);
+        }
+        CaseFile caseFile = EntityDTOConverter.convertToCaseFileEntity(caseFileDTO);
+        caseFile.setPerson(person);
         entityManager.persist(caseFile);
-        return Optional.of(caseFile);
+
+        caseFileDTO.setId(caseFile.getId());
+        return Optional.of(caseFileDTO);
```

Comment:
Consider moving all this logic to Service layer, and make the DAO method only communicate with the DB:

` entityManager.persist(caseFile);`